### PR TITLE
Add support for displaying sample rate and bit depth in UI

### LINKED ISSUE #2296: Add sortable audio properties/quality column
### Describe the solution you'd like
The bitrate column isn't very useful for lossless audio files, but it can be sorted to show higher quality files first.

SoulseekQt has an `Attributes` column that shows bit depth and sample rate for lossless files, and bitrate for lossy files. Unfortunately, the column isn't sortable.

I want the best of both worlds, where the bitrate is always used as the internal sort value, but the value shown in the GUI is similar to SoulseekQt. This allows us to keep displaying high quality files first.

### PR DESCRIPTION
**Pull request by mathiascode**
Closes #2296

This PR doesn't change the bitrate search result filter, but lays the groundwork for adding sample rate and bit depth support to it later.

Prerequisite for #2376

### CODE REVIEW

**Comment by slook** [comment]
On search:

```
Fatal Python error: Segmentation fault

Thread 0x00007f57cd3b6700 (most recent call first):
  File "/home/user/Git/mathiascode/nicotine-plus/pynicotine/slskproto.py", line 2438 in run
  File "/usr/lib/python3.9/threading.py", line 954 in _bootstrap_inner
  File "/usr/lib/python3.9/threading.py", line 912 in _bootstrap

Thread 0x00007f57cdbb7700 (most recent call first):
  File "/home/user/Git/mathiascode/nicotine-plus/pynicotine/cli.py", line 82 in _handle_prompt
  File "/home/user/Git/mathiascode/nicotine-plus/pynicotine/cli.py", line 44 in run
  File "/usr/lib/python3.9/threading.py", line 954 in _bootstrap_inner
  File "/usr/lib/python3.9/threading.py", line 912 in _bootstrap

Thread 0x00007f57d6ad4700 (most recent call first):
  File "/home/user/Git/mathiascode/nicotine-plus/pynicotine/events.py", line 279 in _run_scheduler
  File "/usr/lib/python3.9/threading.py", line 892 in run
  File "/usr/lib/python3.9/threading.py", line 954 in _bootstrap_inner
  File "/usr/lib/python3.9/threading.py", line 912 in _bootstrap

Current thread 0x00007f57d75f8740 (most recent call first):
  File "/home/user/Git/mathiascode/nicotine-plus/pynicotine/gtkgui/search.py", line 868 in add_row_to_model
  File "/home/user/Git/mathiascode/nicotine-plus/pynicotine/gtkgui/search.py", line 850 in append
  File "/home/user/Git/mathiascode/nicotine-plus/pynicotine/gtkgui/search.py", line 757 in add_result_list
  File "/home/user/Git/mathiascode/nicotine-plus/pynicotine/gtkgui/search.py", line 816 in file_search_response
  File "/home/user/Git/mathiascode/nicotine-plus/pynicotine/gtkgui/search.py", line 295 in file_search_response
  File "/home/user/Git/mathiascode/nicotine-plus/pynicotine/events.py", line 225 in emit
  File "/home/user/Git/mathiascode/nicotine-plus/pynicotine/events.py", line 255 in process_thread_events
  File "/home/user/Git/mathiascode/nicotine-plus/pynicotine/gtkgui/application.py", line 683 in on_process_thread_events
  File "/usr/lib/python3/dist-packages/gi/overrides/Gio.py", line 43 in run
  File "/home/user/Git/mathiascode/nicotine-plus/pynicotine/gtkgui/application.py", line 83 in run
  File "/home/user/Git/mathiascode/nicotine-plus/pynicotine/gtkgui/__init__.py", line 101 in run
  File "/home/user/Git/mathiascode/nicotine-plus/pynicotine/__init__.py", line 215 in run
  File "/home/user/Git/mathiascode/nicotine-plus/./nicotine", line 34 in load_module
  File "/home/user/Git/mathiascode/nicotine-plus/./nicotine", line 38 in <module>
Segmentation fault

```

File: pynicotine/slskmessages.py [hunk 1]
```diff
@@ -2514,13 +2514,16 @@ def parse_result_bitrate_length(cls, filesize, attributes):
...
             bitrate = 0
-            h_bitrate = ""
+            h_quality = ""
+
+        elif sample_rate and bit_depth:
+            h_quality = f"{sample_rate / 1000:.4g} kHz / {bit_depth} bit"
 
         else:
-            h_bitrate = f"{bitrate}"
+            h_quality = f"{bitrate} kbps"
```
Comment (slook) [inline pynicotine/slskmessages.py:2523]: kbps -> Kbps

Comment (mathiascode) [reply pynicotine/slskmessages.py:2523]: Maybe it's worth checking what other applications do. I'm just used to kbps for audio.

**Comment by slook** [comment]
Even for lossless files, it is still interesting to know the Bitrate, for example to enter into the Bitrate filter.

Is there any possibility of displaying the bitrate value in the tooltip?

File: pynicotine/slskmessages.py [hunk 2]
```diff
@@ -2490,7 +2490,7 @@ def parse_file_attributes(attributes):
         return bitrate, length, vbr, sample_rate, bit_depth
 
     @classmethod
-    def parse_result_bitrate_length(cls, filesize, attributes):
+    def parse_result_quality_length(cls, filesize, attributes):
```
Comment (slook) [inline pynicotine/slskmessages.py:2493]: `parse_result_attributes`

**Comment by mathiascode** [comment]
> Is there any possibility of displaying the bitrate value (from column 11) in the tooltip?

I don't think it's possible, at least not easily, but I could probably add the bitrate to the file properties dialog.

**Comment by mathiascode** [comment]
Are we happy with the current changes to the transfer list format? There's no going back once this PR is merged.

**Comment by slook** [comment]
Please elaborate on your rationale.

**Comment by mathiascode** [comment]
If we ever plan on doing things with file attributes in the transfer views, like adding columns for them, we need the raw numerical data. The previous bitrate/duration strings can't be formatted in any special way in the GUI once they're stored.

We don't use string keys because they use a lot of memory and disk space (as previously seen with user shares https://github.com/nicotine-plus/nicotine-plus/commit/508782f5dfbd6c20bd5f991aacac2ae24ef7e9a1), it's consistent with user shares, and it's easier to use file attributes directly from the network messages.

**Comment by slook** [comment]
Sounds reasonable.

Where will the originating `search_term` data go?

**Comment by mathiascode** [comment]
> Where will the originating `search_term` data go?

The naive way would be to add a new list item for each row in downloads.json, and a search_term variable to the Transfer object, but I'd have to see how much memory/space it uses. If disk space isn't an issue, but memory is, maybe some dictionary that maps from index to search term, and storing the index in the Transfer object.

**Pull request merged**